### PR TITLE
Fixed mhtml quopri encoding with spaces and tabs (see: #2063)

### PR DESCRIPTION
**Pull request by Raihid**
Saving some space when downloading sites as mhtml.
When a space is encountered at the end of the line, it's encoded as `=20` instead of `<space>=`, but it seems to be OK with Quoted-Printable docs. 
I attach [MHTML file](https://github.com/The-Compiler/qutebrowser/files/618055/qutebrowser_mhtml.zip) generated from qutebrowser.org and UnMHT's version submitted by @stroobandt to compare it with.
Obviously I had to rewrite test .mht files, since they had included all the unnecessary `=20`.

### CODE REVIEW

File: qutebrowser/browser/webkit/mhtml.py [hunk 1]
```diff
@@ -32,6 +32,7 @@
 import email.encoders
 import email.mime.multipart
 import email.message
+from quopri import encodestring
```
Comment (The-Compiler) [inline qutebrowser/browser/webkit/mhtml.py]: Please only import modules (i.e. `import quopri` and use `encodestring.quopri`), as this makes things much more readable IMHO.

File: qutebrowser/browser/webkit/mhtml.py [hunk 2]
```diff
@@ -138,6 +139,22 @@ def _check_rel(element):
     return any(rel in rels for rel in must_have)
 
 
+def _encode_quopri_mhtml(msg):
+    """Encode the message's payload in quoted-printable.
+
+       Substitute for quopri's default 'encode_quopri' method, which needlessly
+       encodes all spaces and tabs, instead of only those at the end on the
+       line.
```
Comment (The-Compiler) [inline qutebrowser/browser/webkit/mhtml.py]: Dedent the description by one, as per [pep 257](https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings)

**Comment by Kingdread** [comment]
LGTM, though I would probably add another unit test that tests the handling of breaking at whitespace/trailing whitespace, since the last character in a line must not be a literal space. Something similar to `test_quotes_printable_umlauts` maybe:

```
def test_quoted_printable_spaces(checker):
    content = b' ' * 100
    writer = mhtml.MHTMLWriter(root_content=content,
                               content_location='localhost',
                               content_type='text/plain')
    writer.write_to(checker.fp)
    checker.expect("""
        Content-Type: multipart/related; boundary="---=_qute-UUID"
        MIME-Version: 1.0

        -----=_qute-UUID
        Content-Location: localhost
        MIME-Version: 1.0
        Content-Type: text/plain
        Content-Transfer-Encoding: quoted-printable

        {}=
        {}=20
        -----=_qute-UUID--
        """.format(' ' * 75, ' ' * 24))
```

We didn't have this before, as every space got encoded anyway.

**Comment by Raihid** [comment]
Yup, that sounds right. Should I just add and test out your code? Or let you do it/write some more tests?

**Comment by The-Compiler** [comment]
It's probably easiest if you just copy that test to your PR

**Comment by The-Compiler** [comment]
Thanks!
